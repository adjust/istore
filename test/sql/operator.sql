SELECT '1=>1, -1=>0'::istore -> -1;
SELECT '1=>1, -1=>3'::istore -> -1;

SELECT '1=>1, -1=>3'::istore ? -1;
SELECT '1=>1, -1=>3'::istore ? 5;

SELECT '1=>1, -1=>3'::istore + '1=>1'::istore;
SELECT '1=>1, -1=>3'::istore + '-1=>-1'::istore;
SELECT '1=>1, -1=>3'::istore + '1=>-1'::istore;

SELECT '1=>1, -1=>3'::istore - '1=>1'::istore;
SELECT '1=>1, -1=>3'::istore - '-1=>-1'::istore;
SELECT '1=>1, -1=>3'::istore - '1=>-1'::istore;

SELECT '1=>3, 2=>2'::istore * '1=>2, 3=>5'::istore;
SELECT '-1=>3, 2=>2'::istore * '-1=>2, 3=>5'::istore;
SELECT '-1=>3, 2=>2'::istore * '-1=>-2, 3=>5'::istore;
