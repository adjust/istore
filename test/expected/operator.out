SELECT '1=>1, -1=>0'::istore -> -1;
 ?column? 
----------
        0
(1 row)

SELECT '1=>1, -1=>3'::istore -> -1;
 ?column? 
----------
        3
(1 row)

SELECT '1=>1, -1=>3'::istore ? -1;
 ?column? 
----------
 t
(1 row)

SELECT '1=>1, -1=>3'::istore ? 5;
 ?column? 
----------
 f
(1 row)

SELECT '1=>1, -1=>3'::istore + '1=>1'::istore;
      ?column?      
--------------------
 "-1"=>"3","1"=>"2"
(1 row)

SELECT '1=>1, -1=>3'::istore + '-1=>-1'::istore;
      ?column?      
--------------------
 "-1"=>"2","1"=>"1"
(1 row)

SELECT '1=>1, -1=>3'::istore + '1=>-1'::istore;
      ?column?      
--------------------
 "-1"=>"3","1"=>"0"
(1 row)

SELECT '1=>1, -1=>3'::istore + 1;
      ?column?      
--------------------
 "-1"=>"4","1"=>"2"
(1 row)

SELECT '1=>1, -1=>3'::istore + 1;
      ?column?      
--------------------
 "-1"=>"4","1"=>"2"
(1 row)

SELECT '1=>1, -1=>3'::istore + 1;
      ?column?      
--------------------
 "-1"=>"4","1"=>"2"
(1 row)

SELECT '1=>1, -1=>3'::istore + 0;
      ?column?      
--------------------
 "-1"=>"3","1"=>"1"
(1 row)

SELECT '1=>1, -1=>3'::istore + 0;
      ?column?      
--------------------
 "-1"=>"3","1"=>"1"
(1 row)

SELECT '1=>1, -1=>3'::istore + 0;
      ?column?      
--------------------
 "-1"=>"3","1"=>"1"
(1 row)

SELECT '1=>1, -1=>3'::istore + -1;
      ?column?      
--------------------
 "-1"=>"2","1"=>"0"
(1 row)

SELECT '1=>1, -1=>3'::istore + -1;
      ?column?      
--------------------
 "-1"=>"2","1"=>"0"
(1 row)

SELECT '1=>1, -1=>3'::istore + -1;
      ?column?      
--------------------
 "-1"=>"2","1"=>"0"
(1 row)

SELECT '1=>1, -1=>3'::istore - '1=>1'::istore;
      ?column?      
--------------------
 "-1"=>"3","1"=>"0"
(1 row)

SELECT '1=>1, -1=>3'::istore - '-1=>-1'::istore;
      ?column?      
--------------------
 "-1"=>"4","1"=>"1"
(1 row)

SELECT '1=>1, -1=>3'::istore - '1=>-1'::istore;
      ?column?      
--------------------
 "-1"=>"3","1"=>"2"
(1 row)

SELECT '1=>1, -1=>3'::istore - 1;
      ?column?      
--------------------
 "-1"=>"2","1"=>"0"
(1 row)

SELECT '1=>1, -1=>3'::istore - 1;
      ?column?      
--------------------
 "-1"=>"2","1"=>"0"
(1 row)

SELECT '1=>1, -1=>3'::istore - 1;
      ?column?      
--------------------
 "-1"=>"2","1"=>"0"
(1 row)

SELECT '1=>1, -1=>3'::istore - 0;
      ?column?      
--------------------
 "-1"=>"3","1"=>"1"
(1 row)

SELECT '1=>1, -1=>3'::istore - 0;
      ?column?      
--------------------
 "-1"=>"3","1"=>"1"
(1 row)

SELECT '1=>1, -1=>3'::istore - 0;
      ?column?      
--------------------
 "-1"=>"3","1"=>"1"
(1 row)

SELECT '1=>1, -1=>3'::istore - -1;
      ?column?      
--------------------
 "-1"=>"4","1"=>"2"
(1 row)

SELECT '1=>1, -1=>3'::istore - -1;
      ?column?      
--------------------
 "-1"=>"4","1"=>"2"
(1 row)

SELECT '1=>1, -1=>3'::istore - -1;
      ?column?      
--------------------
 "-1"=>"4","1"=>"2"
(1 row)

SELECT '1=>3, 2=>2'::istore * '1=>2, 3=>5'::istore;
           ?column?            
-------------------------------
 "1"=>"6","2"=>NULL,"3"=>NULL,
(1 row)

SELECT '-1=>3, 2=>2'::istore * '-1=>2, 3=>5'::istore;
            ?column?            
--------------------------------
 "-1"=>"6","2"=>NULL,"3"=>NULL,
(1 row)

SELECT '-1=>3, 2=>2'::istore * '-1=>-2, 3=>5'::istore;
            ?column?             
---------------------------------
 "-1"=>"-6","2"=>NULL,"3"=>NULL,
(1 row)

SELECT '1=>1, -1=>3'::istore * 1;
      ?column?      
--------------------
 "-1"=>"3","1"=>"1"
(1 row)

SELECT '1=>1, -1=>3'::istore * 1;
      ?column?      
--------------------
 "-1"=>"3","1"=>"1"
(1 row)

SELECT '1=>1, -1=>3'::istore * 1;
      ?column?      
--------------------
 "-1"=>"3","1"=>"1"
(1 row)

SELECT '1=>1, -1=>3'::istore * 0;
      ?column?      
--------------------
 "-1"=>"0","1"=>"0"
(1 row)

SELECT '1=>1, -1=>3'::istore * 0;
      ?column?      
--------------------
 "-1"=>"0","1"=>"0"
(1 row)

SELECT '1=>1, -1=>3'::istore * 0;
      ?column?      
--------------------
 "-1"=>"0","1"=>"0"
(1 row)

SELECT '1=>1, -1=>3'::istore * -1;
       ?column?       
----------------------
 "-1"=>"-3","1"=>"-1"
(1 row)

SELECT '1=>1, -1=>3'::istore * -1;
       ?column?       
----------------------
 "-1"=>"-3","1"=>"-1"
(1 row)

SELECT '1=>1, -1=>3'::istore * -1;
       ?column?       
----------------------
 "-1"=>"-3","1"=>"-1"
(1 row)

